miden-tx/tests/test_miden_atomic_swap.rs

```diff
@@ -0,0 +1,86 @@
+use common::{
+    get_account_with_default_account_code, get_new_key_pair_with_advice_map, MockDataStore,
+};
+use miden_lib::notes::{create_note, Script};
+use miden_objects::{
+    accounts::{Account, AccountId, AccountVault, ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN},
+    assembly::ProgramAst,
+    assets::{Asset, FungibleAsset},
+    Felt,
+};
+use miden_tx::TransactionExecutor;
+use mock::constants::{ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN, ACCOUNT_ID_SENDER};
+use vm_core::StarkField;
+
+mod common;
+
+#[test]
+fn test_atomic_swap_script() {
+    // Create assets
+    let faucet_id = AccountId::try_from(ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN).unwrap();
+    let fungible_asset: Asset = FungibleAsset::new(faucet_id, 100).unwrap().into();
+
+    // Create sender and target account
+    let sender_account_id = AccountId::try_from(ACCOUNT_ID_SENDER).unwrap();
+
+    let target_account_id =
+        AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN).unwrap();
+    let (target_pub_key, target_sk_pk_felt) = get_new_key_pair_with_advice_map();
+    let target_account =
+        get_account_with_default_account_code(target_account_id, target_pub_key.clone(), None);
```

Comment:
If this is the target account, you need to give this second & executing account some assets to swap. If not the test will fail. 

https://github.com/0xPolygonMiden/miden-base/blob/9806a922a3b99948f6254f8ae105904dd4eda7ea/miden-tx/tests/test_miden_wallet.rs#L122-L126